flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/ContaineredTaskManagerParameters.java

```diff
@@ -0,0 +1,182 @@
+
+	private static final long serialVersionUID = -3096987654278064670L;
+	
+	/** Total container memory, in bytes */
+	private final long totalContainerMemoryMB;
+
+	/** Heap size to be used for the Java process */
+	private final long taskManagerHeapSizeMB;
+
+	/** Direct memory limit for the Java process */
+	private final long taskManagerDirectMemoryLimitMB;
+
+	/** The number of slots per TaskManager */
+	private final int numSlots;
+	
+	/** Environment variables to add to the Java process */
+	private final HashMap<String, String> taskManagerEnv;
+
+	
+	public ContaineredTaskManagerParameters(
+		long totalContainerMemoryMB, long taskManagerHeapSizeMB,
+		long taskManagerDirectMemoryLimitMB,
+		int numSlots,
+		HashMap<String, String> taskManagerEnv)
+	{
+		this.totalContainerMemoryMB = totalContainerMemoryMB;
+		this.taskManagerHeapSizeMB = taskManagerHeapSizeMB;
+		this.taskManagerDirectMemoryLimitMB = taskManagerDirectMemoryLimitMB;
+		this.numSlots = numSlots;
+		this.taskManagerEnv = taskManagerEnv;
+	}
+	
+	// ------------------------------------------------------------------------
+
+	public long taskManagerTotalMemoryMB() {
+		return totalContainerMemoryMB;
+	}
+
+	public long taskManagerHeapSizeMB() {
+		return taskManagerHeapSizeMB;
+	}
+
+	public long taskManagerDirectMemoryLimitMB() {
+		return taskManagerDirectMemoryLimitMB;
+	}
+
+	public int numSlots() {
+		return numSlots;
+	}
+
+	public Map<String, String> taskManagerEnv() {
+		return taskManagerEnv;
+	}
+
+
+	// ------------------------------------------------------------------------
+	
+	@Override
+	public String toString() {
+		return "TaskManagerParameters {" +
+			"totalContainerMemory=" + totalContainerMemoryMB +
+			", taskManagerHeapSize=" + taskManagerHeapSizeMB +
+			", taskManagerDirectMemoryLimit=" + taskManagerDirectMemoryLimitMB +
+			", numSlots=" + numSlots +
+			", taskManagerEnv=" + taskManagerEnv +
+			'}';
+	}
+
+	// ------------------------------------------------------------------------
+	//  Factory
+	// ------------------------------------------------------------------------
+	
+	/**
+	 * Computes the parameters to be used to start a TaskManager Java process.
+	 *
+	 * @param config The Flink configuration.
+	 * @param containerMemoryMB The size of the complete container, in megabytes.
+	 * @return The parameters to start the TaskManager processes with.
+	 */
+	public static ContaineredTaskManagerParameters create(
+		Configuration config, long containerMemoryMB, int numSlots)
+	{
+		// (1) compute how much memory we subtract from the total memory, to get the Java memory
+
+		final float memoryCutoffRatio = config.getFloat(
+			ConfigConstants.CONTAINERED_HEAP_CUTOFF_RATIO,
+			ConfigConstants.DEFAULT_YARN_HEAP_CUTOFF_RATIO);
+
+		final int minCutoff = config.getInteger(
+			ConfigConstants.CONTAINERED_HEAP_CUTOFF_MIN,
+			ConfigConstants.DEFAULT_YARN_HEAP_CUTOFF);
+
+		if (memoryCutoffRatio > 1 || memoryCutoffRatio < 0) {
```

Comment:
I suppose it doesn't matter, even 0.9999 will probably fail hard. However, it is not consistent with the error message that states that the ratio should be between 0 and 1.